apps/wallet/src/ui/app/pages/dapp-tx-approval/TransactionSummaryCard.tsx

```diff
@@ -0,0 +1,152 @@
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+import type { CoinsMetaProps } from '_redux/slices/transaction-requests';
+
+import st from './DappTxApprovalPage.module.scss';
+
+const TRUNCATE_MAX_LENGTH = 10;
+const TRUNCATE_PREFIX_LENGTH = 6;
+
+type TransferSummerCardProps = {
+    coinsMeta: CoinsMetaProps[];
+    origin: string;
+    objectIDs: string[];
+    gasEstimate: number | null;
+};
+
+function MiniNFTLink({ id }: { id: string }) {
+    const objectId = useMiddleEllipsis(
+        id,
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+    const nftMeta = useGetNFTMeta(id);
+    return (
+        <>
+            {nftMeta && (
+                <MiniNFT
+                    url={nftMeta.url}
+                    name={nftMeta?.name || 'NFT Image'}
+                    size="xs"
+                />
+            )}
+            <ExplorerLink
+                type={ExplorerLinkType.object}
+                objectID={id}
+                className={st.objectId}
+                showIcon={false}
+            >
+                {objectId}
+            </ExplorerLink>
+        </>
+    );
+}
+
+function CoinMeta({
+    receiverAddress,
+    coinMeta,
+    origin,
+}: {
+    receiverAddress: string;
+    coinMeta: CoinsMetaProps;
+    origin: string;
+}) {
+    const [formatedAmount, symbol] = useFormatCoin(
+        coinMeta.amount ? Math.abs(coinMeta.amount) : 0,
+        coinMeta.coinType
+    );
+    return (
+        <div className={st.content} key={receiverAddress}>
+            <div className={st.row}>
+                <div className={st.label}>Send</div>
+                <div className={st.value}>
+                    {formatedAmount} {symbol}
+                </div>
+            </div>
+
+            <div className={st.row}>
+                <div className={st.label}>To</div>
+                <div className={st.value}>
+                    <div className={st.value}>
+                        <ExternalLink
+                            href={origin}
+                            className={st.origin}
+                            showIcon={false}
+                        >
+                            {new URL(origin || '').host}
+                        </ExternalLink>
+                    </div>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+export function TransactionSummaryCard({
+    objectIDs,
+    coinsMeta,
+    gasEstimate,
+    origin,
+}: TransferSummerCardProps) {
+    const [gasEst, gasSymbol] = useFormatCoin(gasEstimate || 0, GAS_TYPE_ARG);
+
+    return (
+        <SummaryCard>
+            <SummaryCardHeader>Transaction summary</SummaryCardHeader>
+            <SummaryCardContent>
+                {coinsMeta &&
+                    coinsMeta.map((coinMeta, index) => (
+                        <CoinMeta
+                            receiverAddress={coinMeta.receiverAddress}
+                            key={index}
```

Comment:
better use receiverAddress as key or something else rather than the index

Reply:
If a receiver has multiple coinTypes their address will be duplicate

Reply:
Correct the key of [the reduce](https://github.com/MystenLabs/sui/pull/6215/files#diff-e2163eedb634a6ec58750c96512c56e171b94d0e83432e40dddf56dd182de37cR137) `${value.receiverAddress}${value.coinType}` should be better

Reply:
The Key gets removed.

Reply:
But coin meta has the receiverAddress and coinType, so you could use that right?

Reply:
You are right, I'll update it